base_local_planner/src/align_with_path_function.cpp

```diff
@@ -28,11 +29,9 @@ bool AlignWithPathFunction::prepare() {
 }
 
 double AlignWithPathFunction::scoreTrajectory(Trajectory &traj) {
-  if (!isTurningRequired()) {
-    return 0;
-  }
-
-  return std::abs(angles::normalize_angle(current_yaw_diff_ - traj.thetav_ * PREDICTION_TIME));
+  // make spin velocity proportional to delta yaw
+  // spin fast when far away from target yaw and slow down once we get closer
+  return std::abs(std::abs(current_yaw_diff_) / M_PI - std::abs(traj.thetav_) / (max_vel_theta_ + 1e-3));
```

Comment:
please investigate what the actual issue with the current code is; naturally I'd leave the same comments I left in the last PR, which are that this calculation is questionable (i.e. rewards rotating away from the desired heading too) and that `isTurningRequired()` should be kept because e.g. retracting the nose `setXShift(0.0);` is needed to rotate in a narrow corridor.

Reply:
> isTurningRequired() should be kept because e.g. retracting the nose setXShift(0.0); is needed to rotate in a narrow corridor.
I don't get how this is a problem? By removing `isTurningRequired` it's basically the same as making the function always return true. (i.e. as long as we're within goal distance, we rotate). So this shouldn't cause the nose to not be retracted when needed

Reply:
> Rewards rotating away from the desired heading too

It doens't. `current_yaw_diff_` is the current yaw different, not a forward simulated value. The only thing our implementation is scale the rotation speed based on the robot angular distance)